### PR TITLE
Clean tokens in case of idp issuer change

### PR DESCRIPTION
**Pull request by SerhiiBovtriuk**

### CODE REVIEW

File: TDConnectIosSdk/OAuth2Module.swift [hunk 1]
```diff
@@ -164,6 +164,21 @@ open class OAuth2Module: NSObject, AuthzModule, SFSafariViewControllerDelegate {
...
+        handlePossibleIdpChange();
+    }
+    
+    func handlePossibleIdpChange() {
+        if (self.oauth2Session.idToken == nil) {
+            self.oauth2Session.clearTokens();
+            return;
+        }
+        
+        let error = self.validateJwtIdToken(idToken: self.oauth2Session.idToken!)
```
Comment (ulfcarlin) [inline TDConnectIosSdk/OAuth2Module.swift:177]: I suspect that we should only check the issuer ID here (that is, that the issuer ID (OAuth URL) given by the IdProvider matches the issuer ID in the ID token)? Decoding and validating the ID token should only be done once right after receiving the ID token. The reason is that validation includes checking that the ID token has not expired, which happens after about an hour.

Comment (SerhiiBovtriuk) [reply TDConnectIosSdk/OAuth2Module.swift]: That's right, changed implementation.

**Review by ulfcarlin — COMMENTED** [review]
Consider improving the commit message

File: TDConnectIosSdk/OAuth2Module.swift [hunk 2]
```diff
@@ -164,6 +164,33 @@ open class OAuth2Module: NSObject, AuthzModule, SFSafariViewControllerDelegate {
         if (config.wellKnownConfigurationEndpoint != nil) {
             self.fetchWellknownConfig()
         }
+        
+        handlePossibleIdpChange();
+    }
+    
+    func handlePossibleIdpChange() {
+        if (self.oauth2Session.idToken == nil) {
+            self.oauth2Session.clearTokens();
```
[the remark this answers is not in the document]
Comment (SerhiiBovtriuk) [reply TDConnectIosSdk/OAuth2Module.swift:173]: Could be, I will remove the check.

**Review by ulfcarlin — APPROVED** [review]

**Pull request merged**